python/perf-kernels/03-matrix-multiplication-all-types.py

```diff
@@ -205,37 +208,50 @@ def copy_kernel(input_ptr, output_ptr, n_elements, BLOCK_SIZE: tl.constexpr):
 # ---------
 #
 # We can test our custom matrix multiplication operation against a native torch implementation (i.e., rocBLAS).
-@pytest.mark.parametrize("M, N, K, in_dtype, out_dtype",
-[ (*shape, in_dtype, out_dtype)
-    for shape in [(128, 256, 32), (128, 16, 32), (32, 128, 64),
-                  (128, 128, 64), (64, 128, 128), (32, 128, 64),
-                  (64, 64, 32), (32, 32, 128), (128, 128, 64),
-                   (64, 128, 128), (512, 512, 512), (1024, 1024, 1024)]
+def get_x_vals():
+    x_vals = [(1024 * v, 1024 * v, 1024 * v) for v in range (1, 9)]
+
+    x_vals += [(4864, 4096, 8192), (9728, 8192, 65536)]
+
+    return x_vals
+
+@pytest.mark.parametrize("M, N, K, in_dtype, out_dtype, col_a, col_b",
+[ (*shape, in_dtype, out_dtype, col_a, col_b)
+    for shape in get_x_vals()
     for in_dtype, out_dtype in [('fp16', 'fp16'),
                                 ('bf16', 'bf16'),
                                 ('fp16', 'fp32'),
                                 ('fp32', 'fp32'),
                                 ('fp8e4', 'fp16'),
                                 ('fp8e5', 'fp16'),
-                                ('int8', 'int8'),
-                                ('int8', 'int32')]]
+                                #('int8', 'int8'),
+                                ('int8', 'int32')]
+    # Only test k-major tensors because
+    # 1. This is the most preformant config and the current focus
+    # 2. Other case does not work with num_stages=0 (TODO (zhanglx))
+    for col_a in [False]
+    for col_b in [True]]
 )
-def test_correctness(M, N, K, in_dtype, out_dtype):
-    a, a_fp16 = gen_input(M, K, in_dtype, 1, device='cuda')
-    b, b_fp16 = gen_input(K, N, in_dtype, 2, device='cuda')
+def test_correctness(M, N, K, col_a, col_b, in_dtype, out_dtype):
+    a, a_fp16 = gen_input(M, K, in_dtype, col_a, 1, device='cuda')
+    b, b_fp16 = gen_input(K, N, in_dtype, col_b, 2, device='cuda')
     # Allocates output.
     tl_out_dtype = name_to_tl_types[out_dtype]
-    c = torch.empty((M, N), device=a.device, dtype=tl_to_torch_types[tl_out_dtype])
+    torch_out_dtype = tl_to_torch_types[tl_out_dtype]
+    c = torch.empty((M, N), device=a.device, dtype=torch_out_dtype)
     matmul(a, b, c, activation="")
-    torch_output = torch.matmul(a_fp16, b_fp16)
-    print(f"triton_output={c}")
-    print(f"torch_output={torch_output}")
+    if in_dtype == 'fp8e4' or in_dtype == 'fp8e5' or in_dtype == 'int8':
+        # For f8 and int8 inputs, use fp16 for torch.matmul
+        torch_output = torch.matmul(a_fp16, b_fp16)
+    else:
+        torch_output = torch.matmul(a, b)
+    #print(f"triton_output={c}")
+    #print(f"torch_output={torch_output}")
     rtol = 0 if torch.version.hip is None else 1e-2
-    if torch.allclose(c.to(torch.float16), torch_output, atol=5e-2, rtol=rtol):
-        print("✅ Triton and Torch match")
+    if in_dtype == 'int8':
+        torch.testing.assert_close(c.to(torch.float16), torch_output, atol=5e-2, rtol=rtol)
```

Comment:
If we use a smaller K and do not use splitK, maybe we can use smaller tolerances. 
For `int8` data type, can we cast to int32 type for comparison, so we can use 0 as tolerance.

Reply:
> If we use a smaller K and do not use splitK, maybe we can use smaller tolerances.

I assume you mean smaller BLOCK_K. 
We cannot pick the tiling and parameter here. They are picked by the autotuner. We want to check the correctness of the configs that is used in the benchmark.

> For int8 data type, can we cast to int32 type for comparison, so we can use 0 as tolerance

Let me give it a shot

Reply:
> For int8 data type, can we cast to int32 type for comparison, so we can use 0 as tolerance

It doesn't help. 
For int8 input and int8 output, it seems the precision is lost a lot.